main.go

```diff
@@ -143,6 +176,15 @@ func main() {
 	case "ospatch":
 		ospatchog.Run(ctx, make(chan struct{}))
 		return
+	case "agentendpoint":
```

Comment:
Shouldn't this be in addition to "ospatch" or "policies"? I thought using the agent endpoint was an implementation detail of those features even if it's a global switch.

Reply:
We will need to discuss how we want to gate this (or enabled/disable certain features) as it is very different than in the past. This argument (and all these args) are meant for debug running of the agent, the way the endpoint works is that the agent just connects and listens for tasks, there is no additional flag needed to run a patch task or an exec task. Maybe the name is misleading as this doesn't so much have to do with the agentendpoint as it does with WaitForTaskNotification. The args are not meant to be a stable API so we can rethink this in the future.
Because all this is alpha right now I will block the endpoint connection using our current patch alpha flag and not have one for exec separately as we have that tied to patch in the API. In the future if we are to add additional tasks we can decide if we want to block them via a flag on the agent while they are in alpha, or just use the fact that they are alpha and the agent needs to be told to use the alpha API and so all alpha features will be enabled if that is set.

Also the "ospatch" arg here will be removed as that only corresponds to the current patch workflow and not the new one. "guestpolicies" are agent driven so those will stay and we will just move to using the endpoint calls.